ocs_ci/utility/baremetal.py

```diff
@@ -0,0 +1,227 @@
+import logging
+
+import pyipmi
+import pyipmi.interfaces
+
+from ocs_ci.ocs import constants, defaults
+from ocs_ci.ocs.constants import VM_POWERED_OFF, VM_POWERED_ON
+from ocs_ci.ocs.exceptions import UnexpectedBehaviour
+from ocs_ci.ocs.node import wait_for_nodes_status
+from ocs_ci.ocs.ocp import OCP, wait_for_cluster_connectivity
+from ocs_ci.utility.retry import retry
+from ocs_ci.utility.utils import TimeoutSampler, load_auth_config, exec_cmd
+from tests import helpers
+
+logger = logging.getLogger(__name__)
+
+
+class BAREMETAL(object):
+    """
+    wrapper for Baremetal
+    """
+
+    def __init__(self):
+        """
+        Initialize the variables required
+
+        """
+        self.mgmt_details = load_auth_config()['ipmi']
+
+    def get_ipmi_ctx(self, host, user, password):
+        """
+        Function to get ipmi handler
+        Args:
+            host (str): Host mgmt address
+            user (str): User Name for accessing mgmt console
+            password (str): Password for accessing mgmt console
+
+        Returns (object): ipmi handler
+
+        """
+        interface = pyipmi.interfaces.create_interface('ipmitool',
+                                                       interface_type=defaults.IPMI_INTERFACE_TYPE)
+        ipmi = pyipmi.create_connection(interface)
+        ipmi.session.set_session_type_rmcp(host, port=defaults.IPMI_RMCP_PORT)
+        ipmi.session.set_auth_type_user(user, password)
+        ipmi.session.establish()
+        ipmi.target = pyipmi.Target(ipmb_address=defaults.IPMI_IPMB_ADDRESS)
+        return ipmi
+
+    def get_power_status(self, ipmi_ctx):
+        """
+        Get BM Power status
+
+        Args:
+            ipmi_ctx (object) : Ipmi host handler
+
+        """
+        chassis_status = ipmi_ctx.get_chassis_status()
+        return VM_POWERED_ON if chassis_status.power_on else VM_POWERED_OFF
+
+    @retry(UnexpectedBehaviour, tries=4, delay=3, backoff=1)
+    def verify_machine_is_down(self, baremetal_machine):
+        """
+        Verifiy Baremetal machine is completely power off
+
+        Args:
+            baremetal_machine (list): BM objects
+
+        Raises:
+            UnexpectedBehaviour: If baremetal machine is still up
+
+        """
+        for node in baremetal_machine:
+            result = exec_cmd(cmd=f"ping {node.name} -c 100", ignore_error=True)
```

Comment:
is it possible maybe to ping the server without `-c` (count) and use a timeout sampler logic? That way we can specify the timeout in seconds

Reply:
will try to make use of timeout sampler

Reply:
Done